### ISSUE TITLE
Fatal Exception: java.lang.NullPointerException: getFirebaseInstallationId() must not be null

### ISSUE THREAD

**Issue by wahyupermadie** [firebase/firebase-android-sdk#5765]
### [READ] Step 1: Are you in the right place?

Issues filed here should be about bugs in __the code in this repository__. 
If you have a general question, need help debugging, or fall into some
other category use one of these other channels:

  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Iguana Build #AI-232.10227.8.2321.11479570
  * Firebase Component: Crashlytics
  * Component version: 
  fireBaseBomVersion = '31.4.0'
  firebaseCrashlyticsGradleVersion = "2.9.2"
  firebaseCrashlyticsNdkVersion = '18.3.6'
  
### [REQUIRED] Step 3: Describe the problem
In production app we got this kind of error after updating the firebase version, we update it for the Crash Free Sesson on Firebase Crashlytics.
```
Fatal Exception: java.lang.NullPointerException: getFirebaseInstallationId() must not be null
       at com.google.firebase.sessions.SessionFirelogPublisherImpl$logSession$1.invokeSuspend(SessionFirelogPublisher.kt:73)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at com.google.firebase.concurrent.CustomThreadFactory.lambda$newThread$0(CustomThreadFactory.java:47)
       at java.lang.Thread.run(Thread.java:919)
```
#### Steps to reproduce:

We not able to reproduce locally, only seen the issue on firebase crashlytics report

#### Relevant Code:

```
// TODO(you): code here to reproduce the problem
```

**Comment by gaoxuefeng** [comment]
I also encountered this problem , only one user reported this issue
**libs:**
googleService = "4.4.0"
crashlyticsGradle = "2.9.9"
perfPluginGradle = "1.4.2"
firebaseBoM = "32.7.0"
firebaseAnalytics = "21.5.0"
firebaseCrashlytics = "18.6.0"
firebasePerformance = "20.5.1"
**error:**
SessionFirelogPublisherImpl$logSession$1.invokeSuspend
java.lang.NullPointerException - getFirebaseInstallationId() must not be null
**stack:**
com.google.firebase.sessions.SessionFirelogPublisherImpl$logSession$1.invokeSuspend (SessionFirelogPublisher.kt:73)
kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith (ContinuationImpl.kt:33)
kotlinx.coroutines.DispatchedTask.run (DispatchedTask.kt:108)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
com.google.firebase.concurrent.CustomThreadFactory.lambda$newThread$0 (CustomThreadFactory.java:47)
java.lang.Thread.run (Thread.java:929)
**mobile:**
brand：Cmdc：SP200
vesion：Android 10

**Comment by komissarovdenis** [comment]
Hey! Same here. 
Currently we are seeing a massive crash peak. During the week of gradual deployment, we received more than **500** crashes (**400** users). Various devices, various Android versions.

```
            firebaseAnalytics           : '21.5.1',
            firebaseCrashlytics       : '18.6.2',
            firebasePerf                   : '20.5.2',
            firebaseMessaging        : '23.4.1',
            firebaseInstallations      : '17.2.0',
            firebaseCommon           : '20.4.2',
```

**Comment by lehcar09** [comment]
Hi there! Thank you for reaching out and reporting the issue. I tried reproducing the issue using our, however, I wasn't able to encounter any exception. Can you share an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) to help us investigate the issue? 

Aside from that, does the issue occur on certain devices and/or android versions?

**Comment by wahyupermadie** [comment]
@lehcar09 I'm not able to reproduce it locally, based on data mostly affected on below android 10 and most affected device is Xiaomi. this issue started once we bump the bom version to make crash free session available

**Comment by komissarovdenis** [comment]
@lehcar09 we don't know how to reproduce it. It looks pretty random (maybe a race condition somewhere in the library). 
This issue is 100% not a device/os specific.
<img width="552" alt="image" src="https://github.com/firebase/firebase-android-sdk/assets/6702649/81b6daef-88d8-433e-9f10-aa9ec9698fb0">
<img width="554" alt="image" src="https://github.com/firebase/firebase-android-sdk/assets/6702649/0239163a-7283-4f10-97d1-6e6ea969dc14">

**Comment by tomachi** [comment]
We also started having this issue now after version upgrade.
We had:
'com.google.firebase:firebase-analytics:21.3.0'
'com.google.firebase:firebase-messaging:23.2.1'
'com.google.firebase:firebase-common-ktx:20.4.0'
'com.google.firebase:firebase-crashlytics:18.4.3'

And upgraded to:
'com.google.firebase:firebase-analytics:21.5.0'
'com.google.firebase:firebase-messaging:23.4.0'
'com.google.firebase:firebase-common-ktx:20.4.2'
'com.google.firebase:firebase-crashlytics:18.6.1'

Fatal Exception: java.lang.NullPointerException: getFirebaseInstallationId() must not be null
       at com.google.firebase.sessions.SessionFirelogPublisherImpl$logSession$1.invokeSuspend(SessionFirelogPublisher.kt:71)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at com.google.firebase.concurrent.CustomThreadFactory.lambda$newThread$0(CustomThreadFactory.java:47)
       at java.lang.Thread.run(Thread.java:919)

![Screenshot 2024-03-13 at 11 00 59](https://github.com/firebase/firebase-android-sdk/assets/302540/59c9f75c-0202-47bc-9dd2-5f8dc24c2327)
![Screenshot 2024-03-13 at 11 01 06](https://github.com/firebase/firebase-android-sdk/assets/302540/0ac20d11-b2d2-48ca-8733-45b94e7b0099)

**Comment by mrober** [comment]
This was fixed in the latest version of Firebase Sessions. But it looks like the latest version of Crashlytics has not picked it up yet. I will make sure it is picked up in the next release. For now, you can work around this by manually setting the sessions dependency like so:

```kotlin
// This can be removed after the next release of the Crashlytics SDK.
implementation 'com.google.firebase:firebase-sessions:1.2.2'
```

**Comment by wahyupermadie** [comment]
Hi @mrober thanks for the confirmation, btw is there any way we can try rn to resolve this until the official release ?

**Comment by mrober** [comment]
I am not sure why it only happens on some devices. The change I mentioned above forces validation of the fid in logic, this will be included in the next release.

I also made another change #5784 to handle when the Firebase installations api returns a `Task` of null for whatever reason, but it is too late to get this change into the release. I expect the above change is enough, but was not able to reproduce the issue locally to validate.

**Comment by wahyupermadie** [comment]
thank you @mrober , please inform us once it release thank you

**Comment by aidoTank** [comment]
The logs show that the problematic device is API level <= 27 @mrober

**Comment by wahyupermadie** [comment]
@mrober Hi, want to ask when will this changes release ? do we have eta for this ? thank you

**Comment by lehcar09** [comment]
Hi folks!, the fix (PR #5716) for this is included in the recent release of [BOM 32.8.0](https://firebase.google.com/support/release-notes/android#2024-03-21). That said, Ill be closing this issue now. Let us know if the issue persists and we can re-open the issue. Thanks!

**Comment by wahyupermadie** [comment]
thank you @mrober @lehcar09

**Comment by tomachi** [comment]
We updated firebase-session to 1.2.2 and we are still crashing (we are not using BOM).

> Fatal Exception: java.lang.NullPointerException: getFirebaseInstallationId() must not be null
>        at com.google.firebase.sessions.SessionFirelogPublisherImpl$logSession$1.invokeSuspend(SessionFirelogPublisher.kt:71)
>        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
>        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
>        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
>        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
>        at com.google.firebase.concurrent.CustomThreadFactory.lambda$newThread$0(CustomThreadFactory.java:47)
>        at java.lang.Thread.run(Thread.java:923)

**Comment by tomachi** [comment]
ohh, I see it might be fixed in 1.2.3. We'll give it a try. Don't reopen yet.
